src/main/java/net/fabricmc/installer/client/ClientInstaller.java

```diff
@@ -51,7 +52,7 @@ public static String install(Path mcDir, String gameVersion, String loaderVersio
 		Files.deleteIfExists(dummyJar);
 		Files.createFile(dummyJar);
 
-		URL profileUrl = new URL(Reference.getMetaServerEndpoint(String.format("v2/versions/loader/%s/%s/profile/json", gameVersion, loaderVersion)));
+		URL profileUrl = new URL(Reference.getMetaServerEndpoint(String.format("v2/versions/loader/%s/%s/profile/json", gameVersion, loaderVersion.name)));
```

Comment:
The client will need more work to support custom versions. It will need to copy the custom loader jar into libraries and then make up the json file. This can be done later.